src/store.mjs

```diff
@@ -328,38 +327,62 @@ export async function posts(
 }
 
 export async function leaves(trie, from, amount, parser, startDatetime) {
-  let pointer = 0;
   const nodes = [];
-  const onFound = (_, node, key, walkController) => {
-    if (Number.isInteger(amount) && nodes.length >= amount) return;
 
-    if (node instanceof LeafNode) {
-      pointer += 1;
-      if (Number.isInteger(from) && pointer <= from) return;
+  let pointer = 0;
+  for await (const [node] of walkTrieDfs(trie, trie.root(), [])) {
+    if (Number.isInteger(amount) && nodes.length >= amount) {
+      break;
+    }
 
-      const value = decode(node.value());
-      if (parser) {
-        const parsed = parser(value);
-        if (parsed.timestamp < startDatetime) return;
+    pointer++;
+    if (Number.isInteger(from) && pointer <= from) {
+      continue;
+    }
 
-        nodes.push(parsed);
-      } else {
-        nodes.push(value);
+    const value = decode(node.value());
+    if (parser) {
+      const parsed = parser(value);
+      if (parsed.timestamp < startDatetime) {
+        continue;
       }
-    } else if (node instanceof BranchNode || node instanceof ExtensionNode) {
-      walkController.allChildren(node, key);
-    }
-  };
 
-  try {
-    await trie.walkTrie(trie.root(), onFound);
-  } catch (err) {
-    if (err.toString().includes("Missing node in DB")) {
-      log("leaves: Didn't find any nodes");
-      return nodes;
+      nodes.push(parsed);
     } else {
-      throw err;
+      nodes.push(value);
     }
   }
+
   return nodes;
 }
+
+/**
+ * @param {Trie} trie
+ * @param {Buffer} nodeRef
+ * @param {number[]} key
+ */
+async function* walkTrieDfs(trie, nodeRef, key) {
+  const node = await trie.lookupNode(nodeRef);
+
+  if (node instanceof LeafNode) {
+    yield [node, key];
+    return;
+  }
+
+  if (node instanceof ExtensionNode) {
+    const keyExtension = node.key();
+    const childKey = key.concat(keyExtension);
+    const childRef = node.value();
+    yield* walkTrieDfs(trie, childRef, childKey);
+  } else if (node instanceof BranchNode) {
+    for (let i = 0; i < 16; i++) {
```

Comment:
If we wanted to do descending timestamp order, we can just reverse the order of iteration here.

Reply:
nice, @matallo was interested in that